src/main/java/com/gmail/nossr50/skills/repair/RepairManager.java

```diff
@@ -128,6 +133,34 @@ public void handleRepair(ItemStack item) {
 
         // toRemove should be refreshed before the event call.
         toRemove = inventory.getItem(inventory.first(repairMaterial)).clone();
+
+        // Check if we allow enchanted materials to be used to repair objects.
+        // (Servers may provide enchanted items that don't follow their intended use)
+        if (!mcMMO.p.getAdvancedConfig().getAllowEnchantedRepairMaterials()) {
+
+            // See if our proposed item is even enchanted in the first place.
+            if (toRemove.getEnchantments().size() > 0) {
+
+                // Lots of array sorting to gather a possible list of items not including the enchanted item.
+                List<ItemStack> possibleMaterials = Arrays.stream(inventory.getContents())
+                        .filter(Objects::nonNull)
+                        .filter(p -> p.getType() == repairMaterial)
+                        .filter(p -> p.getEnchantments().size() == 0).toList();
+
+                // Fail out with "you need material" if we don't find a suitable alternative.
+                if (possibleMaterials.size() == 0) {
```

Comment:
A general tip:
`x.isEmpty()` and `!x.isEmpty()` are generally better suitable instead of `x.size() == 0` or `x.size() > 0`.
They are simply more descriptive of the actual check you want to perform.